test/utils/test_utils.py

```diff
@@ -127,7 +108,15 @@ def test_get_prox_and_regularization_bad_shape(regularization, lam):
     "regularization", ["weighted_l0", "weighted_l1", "weighted_l2"]
 )
 @pytest.mark.parametrize(
-    "lam", [np.array([[1, 2]]), np.array([1, 2, 3]), np.array([[1, 2, 3]]).T, 1]
+    "lam",
+    [
+        np.array([[1, 2]]),
+        np.array([1, 2, 3]),
+        np.array([[1, 2, 3]]).T,
+        1,
+        np.array([1]),
+        np.array([[1]]),
+    ],
```

Comment:
As above - I think we only need to test two cases (a) a scalar, and (b) an array that is of broadcastable shape (because we check shape explicitly to prevent broadcast)